File: src/main/java/lana/thingService/util/objectmapper/ObjectMapperImpl.java
package lana.thingService.util.objectmapper;

import org.modelmapper.ModelMapper;

public class ObjectMapperImpl implements ObjectMapper {
    private final ModelMapper modelMapper = new ModelMapper();

    @Override
    public <T> T map(Object source, Class<T> destinationType) {
        return modelMapper.map(source, destinationType);
    }

    @Override
    public void map(Object source, Object destination) {
        modelMapper.map(source, destination);
    }
}

File: src/main/java/lana/thingService/util/objectmapper/ObjectMapperImpl.kt
package lana.thingService.util.objectmapper

import org.modelmapper.ModelMapper

class ObjectMapperImpl : ObjectMapper {
    private val modelMapper = ModelMapper()

    override fun <T> map(source: Any, destinationType: Class<T>): T {
        return modelMapper.map(source, destinationType)
    }

    override fun map(source: Any, destination: Any) {
        modelMapper.map(source, destination)
    }
}
